common/cpu/intel/tiger-lake/default.nix

```diff
@@ -0,0 +1,15 @@
+{ config, lib, ... }:
+let
+  hasXeDriver = lib.versionAtLeast config.boot.kernelPackages.kernel.version "6.8";
+in
+{
+  imports = [ ../. ];
+  config = lib.mkMerge [
+    (lib.mkIf hasXeDriver {
+      # Favor xe driver over i915 if it exists
+      hardware.intelgpu.loadXeInInitrd = true;
+      hardware.intelgpu.loadInInitrd = false;
+    })
+    (lib.mkIf (!hasXeDriver) { boot.kernelParams = [ "i915.enable_guc=3" ]; })
```

Comment:
This I would move also to the gpu module and do a `hardware.intelgpu.driver == "xe"` comparison.

Reply:
I think we can't actually do that because "i915.enable_guc=3" is only applicable to tiger-lake, so it has to stay here. You can see other chipsets will have other `enable_guc` settings, and don't support the value 3. Also only 6.8 or newer kernels support the xe driver, but otherwise people will still use i915 on the tiger-lake. I think I've addressed this now though in the latest commit, but switching the tests around a bit and being more clear about the kernel version requirements.